software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+      put "on userRedo" into xX["write"];    devLog xX
+   put true into tDo["redo"]
+   changeRequest tDo,sMain,sNowDoc
+   refreshDraw
+   refreshDisplay
+end userRedo
+
+on instrButtMouseUp
+   put "on instrButtMouseUp" into xX["write"];    devLog xX
+   -- need to save so there's a place to put stuff
+   if sMain[sNowDoc][0]["filepath"] is empty then
+      saveAsButtMouseUp
+      if sMain[sNowDoc][0]["filepath"] is empty then exit instrButtMouseUp
+   end if
+   -- adjList already called on every change
+   topoSort sMainShort,sTopoSort,sNotVisited,sNoteList
+   --   put  writeInstructions(sMain,sMainShort,sTopoSort,sNotVisited) into tInstr
+   put writeInstructionsHTML(sMain,sMainShort,sTopoSort,sNotVisited) into tInstr
+   --   set the clipboardData["text"] to tInstr
+   --   answer tInstr
+   -- create or overwrite the instructions.html file
+   put sMain[sNowDoc][0]["filepath"] into tPath -- path for the *.how file
+   set the itemdelimiter to slash
+   put "instructions.html" into item -1 of tPath
+   put tInstr into URL ("file:" & tPath)
+   launch url ("file:" & tPath)
+end instrButtMouseUp
+
+on resoButtMouseUp
+   put "on resoButtMouseUP" into xX["write"];    devLog xX
+   -- adjList already called on every change
+   --   topoSort sMainShort,sTopoSort,sNotVisited,sNoteList
+   supplyDemand sMainShort,sScopeSketch,sResources,sSomethingChanged,sNotVisited
+   --   put summarizeResources(sMain,sScopeSketch,sResources,sNowDoc) into tRes
+   put summarizeResourcesHTML(sMain,sMainShort,sScopeSketch,sResources,sNowDoc) into tRes
+   --   answer tRes   
+   -- create or overwrite the instructions.html file
+   put sMain[sNowDoc][0]["filepath"] into tPath -- path for the *.how file
+   set the itemdelimiter to slash
+   put "resources.html" into item -1 of tPath
+   put tRes into URL ("file:" & tPath)
+   launch url ("file:" & tPath)
+end resoButtMouseUp
+
+on saveButtMouseUp
+   put "on saveButtMouseUP" into xX["write"];    devLog xX
+   if sMain[sNowDoc][0]["filepath"] is empty then
+      saveAsButtMouseUp
+   else
+      --      dispatch "cmdJsonFromArray" to stack "easyJSON" with sMain[sNowDoc]
+      --      put cmdJsonFromArray(sMain[sNowDoc]) into tResult
+      put jsonFromArray(sMain[sNowDoc]) into tResult
+      --      put the result into URL ("file:" & sMain[sNowDoc][0]["filepath"])
+      put tResult into URL ("file:" & sMain[sNowDoc][0]["filepath"])
+      put "saved" into sMain[sNowDoc][0]["saveStatus"]
+      set itemdelimiter to slash
+      put item -1 of sMain[sNowDoc][0]["filepath"] into tForShow
+      replace ".how" with "" in tForShow
+      put tForShow into field "filenameLabel" stack "HowstrBeta"
+   end if
+end saveButtMouseUp
+
+on loadButtMouseUp
+   -- on the main stack; as in load a Howstr *.how file
+   put "on loadButtMouseUp" into xX["write"];    devLog xX
+   if sMain[sNowDoc][0]["saveStatus"] is "unsaved" then
+      answer "Load new without saving current?" with "Close current" or "Cancel"
```

Comment:
I would love to talk about this! I couldn't figure out how to do it back when I was starting so I just worked on the one-doc-at-a-time approach.